lib/camp.js

```diff
@@ -768,18 +783,22 @@ function pathLikeUnit(serverMethod, statusCode) {
           for (var key in matched) {
             req.data[key] = matched[key];
           }
-          if (callbacks[cbindex] !== undefined) {
-            callbacks[cbindex](req, res);
+          var cb = callbacks[cbindex];
+          var method = req.method.toLowerCase();
+          if (cb instanceof Function) {
+            cb(req, res);
+          } else if (cb instanceof Object && cb[method] instanceof Function) {
```

Comment:
This polymorphism is not ideal. Instead, I would recommend to make `paths` an object containing all the filters, eg. `{path: new Path('/' + path), methods: […]}`.

Obviously, you'd have to change the name `paths` to something more appropriate. Maybe `filters`?